fw/vhost.c

```diff
@@ -1336,9 +1373,20 @@ tfw_location_init(TfwLocation *loc, tfw_match_t op, const char *arg,
 	loc->nipdef_sz = 0;
 	loc->hdrs_pool = pool;
 	loc->mod_hdrs[TFW_VHOST_HDRMOD_REQ].hdrs =
-			(TfwHdrModsDesc *)(loc->nipdef + TFW_NIPDEF_ARRAY_SZ);
+		(TfwHdrModsDesc *)(loc->nipdef + TFW_NIPDEF_ARRAY_SZ);
+
+	loc->mod_hdrs[TFW_VHOST_HDRMOD_REQ].spec_hdrs =
+		(TfwHdrModsDesc **)(loc->mod_hdrs[TFW_VHOST_HDRMOD_REQ].hdrs
+				    + TFW_USRHDRS_ARRAY_SZ);
+
 	loc->mod_hdrs[TFW_VHOST_HDRMOD_RESP].hdrs =
-			loc->mod_hdrs[TFW_VHOST_HDRMOD_REQ].hdrs + TFW_USRHDRS_ARRAY_SZ;
+		(TfwHdrModsDesc *)(loc->mod_hdrs[TFW_VHOST_HDRMOD_REQ].spec_hdrs
+				   + TFW_HTTP_HDR_RAW);
+
+	loc->mod_hdrs[TFW_VHOST_HDRMOD_RESP].spec_hdrs =
+		(TfwHdrModsDesc **)(loc->mod_hdrs[TFW_VHOST_HDRMOD_RESP].hdrs
+				    + TFW_USRHDRS_ARRAY_SZ);
```

Comment:
@RomanBelozerov I think it makes sense to test in https://github.com/tempesta-tech/tempesta-test/pull/414 a configuration to modify 64 headers to test the configuration code.

Reply:
@krizhanovsky Seems configuration works fine, but when we use such configuration in tests we have problems in another places that covered by few issues.

Reply:
Yes. You are right, this is `TestManyRequestHeaders`.